### PR TITLE
improve performance of string compare

### PR DESCRIPTION
**Pull request by badboynt1**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:
use three dword instruction to compare varlena
10x faster than before

### CODE REVIEW

File: pkg/sql/plan/function/baseTemplate.go [hunk 1]
```diff
@@ -1609,6 +1609,54 @@ func opBinaryBytesBytesToFixedWithErrorCheck[Tr types.FixedSizeTExceptStrType](
...
+	v1, v2 := parameters[0], parameters[1]
+	if v1.IsConstNull() || v2.IsConstNull() {
+		nulls.AddRange(rsVec.GetNulls(), 0, uint64(length))
+		return nil
+	}
+	col1, _ := vector.MustVarlenaRawData(v1)
+	col2, _ := vector.MustVarlenaRawData(v2)
+
+	if v1.IsConst() && v2.IsConst() {
+		p1 := col1[0].UnsafePtr()
```
Comment (fengttt) [inline pkg/sql/plan/function/baseTemplate.go:1626]: Do you really need unsafe pointe?      Even if you really have to this should be part of vector impl and should not escape the vector package.

File: pkg/sql/plan/function/baseTemplate.go [hunk 2]
```diff
@@ -1609,6 +1609,54 @@ func opBinaryBytesBytesToFixedWithErrorCheck[Tr types.FixedSizeTExceptStrType](
...
+		nulls.Or(parameters[0].GetNulls(), parameters[1].GetNulls(), rsVec.GetNulls())
+	} else if v1.IsConst() {
+		p1 := col1[0].UnsafePtr()
+		for i := 0; i < length; i++ {
+			p2 := col2[i].UnsafePtr()
+			rss[i] = *(*[3]int64)(p1) == *(*[3]int64)(p2)
+		}
+		nulls.Or(rsVec.GetNulls(), parameters[1].GetNulls(), rsVec.GetNulls())
+	} else {
+		p2 := col2[0].UnsafePtr()
```
Comment (fengttt) [inline pkg/sql/plan/function/baseTemplate.go:1650]: Save here.    This loop if you have to go into vector details to get max perf, the following code belongs to the vector package.

File: pkg/sql/plan/function/baseTemplate.go (hunk 1 above)
Comment (badboynt1) [reply pkg/sql/plan/function/baseTemplate.go]: fixed

File: pkg/sql/plan/function/baseTemplate.go (hunk 2 above)
Comment (badboynt1) [reply pkg/sql/plan/function/baseTemplate.go]: fixed

**Review by m-schen — APPROVED** [review]

**Review by aunjgr — APPROVED** [review]

**Pull request merged**